ckanext/dcatapchharvest/dcat_helpers.py

```diff
@@ -167,38 +169,161 @@ def get_frequency_values():
+
+    def _get_license_literal(self, graph, license_ref):
+        for license_pref_label in graph.objects(subject=license_ref,
+                                                predicate=SKOSXL.prefLabel):
+            try:
+                return next(graph.objects(subject=license_pref_label,
+                                          predicate=SKOSXL.literalForm))
+            except StopIteration:
+                continue
+        return None
+
+    def _process_graph(self, graph):
+        license_ref_literal_mapping = {}
+        license_homepages_literal_mapping = {}
+        license_homepage_ref_mapping = {}
+
+        for ogdch_license_ref in graph.subjects(predicate=RDF.type,
+                                                object=SKOS.Concept):
+            license_homepage = self._get_license_homepage(graph,
+                                                          ogdch_license_ref)
+            license_literal = self._get_license_literal(graph,
+                                                        ogdch_license_ref)
+
+            license_homepages_literal_mapping[unicode(license_homepage)] = \
+                unicode(license_literal)
+            license_ref_literal_mapping[unicode(ogdch_license_ref)] = \
+                unicode(license_literal)
+            license_homepage_ref_mapping[unicode(license_homepage)] = \
+                unicode(ogdch_license_ref)
+
+        return (license_homepages_literal_mapping,
+                license_ref_literal_mapping, license_homepage_ref_mapping)
+
+    def _get_license_values(self):
+        if self._license_cache is None:
+            try:
+                g = Graph()
+                self._bind_namespaces(g)
+                self._parse_graph(g)
+
+                (license_homepages_literal_mapping,
+                 license_ref_literal_mapping,
+                 license_homepage_ref_mapping) = self._process_graph(g)
+
+                self._license_cache = (license_homepages_literal_mapping,
+                                       license_ref_literal_mapping,
+                                       license_homepage_ref_mapping)
+            except Exception as e:
+                raise RuntimeError("Failed to load license values: %s"
+                                   % e)
+        return self._license_cache
+
+    def get_license_ref_uri_by_name(self, vocabulary_name):
+        _, license_ref_literal_vocabulary, _ = self._get_license_values()
+        return next((key for key, value in
+                     license_ref_literal_vocabulary.items()
+                     if unicode(vocabulary_name) == value),
+                    None)
+
+    def get_license_ref_uri_by_homepage_uri(self, vocabulary_name):
+        _, _, license_homepage_ref_vocabulary = self._get_license_values()
+        return next((value for key, value in
+                     license_homepage_ref_vocabulary.items()
+                     if unicode(vocabulary_name) == key),
+                    None)
```

Comment:
Here, and in `get_license_name_by_ref_uri` and `get_license_name_by_homepage_uri`, you could just do this:

```
return license_homepage_ref_vocabulary.get(unicode(vocabulary_name)
```